app/src/main/java/org/oppia/android/app/customview/ContinueButtonView.kt

```diff
@@ -0,0 +1,129 @@
+package org.oppia.android.app.customview
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.animation.AnimationUtils
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import org.oppia.android.R
+import org.oppia.android.app.model.EphemeralState
+import org.oppia.android.app.player.state.itemviewmodel.ContinueInteractionViewModel
+import org.oppia.android.app.utility.LifecycleSafeTimerFactory
+import org.oppia.android.app.view.ViewComponentFactory
+import org.oppia.android.app.view.ViewComponentImpl
+import org.oppia.android.domain.exploration.ExplorationProgressController
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import org.oppia.android.util.platformparameter.EnableContinueButtonAnimation
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.oppia.android.util.system.OppiaClock
+import javax.inject.Inject
+
+class ContinueButtonView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = R.style.StateButtonActive
+) : androidx.appcompat.widget.AppCompatButton(context, attrs, defStyleAttr) {
+
+  @field:[Inject EnableContinueButtonAnimation]
+  lateinit var enableContinueButtonAnimation: PlatformParameterValue<Boolean>
+
+  @Inject
+  lateinit var fragment: Fragment
+
+  @Inject
+  lateinit var oppiaClock: OppiaClock
+
+  @Inject
+  lateinit var explorationProgressController: ExplorationProgressController
+
+  @Inject
+  lateinit var lifecycleSafeTimerFactory: LifecycleSafeTimerFactory
+
+  private lateinit var viewModel: ContinueInteractionViewModel
+  private var isAnimationTimerFinished = false
+
+  val animateContinueButton = MutableLiveData(false)
+
+  private val ephemeralStateLiveData: LiveData<AsyncResult<EphemeralState>> by lazy {
+    explorationProgressController.getCurrentState().toLiveData()
+  }
+
+  private fun subscribeToCurrentState() {
+    ephemeralStateLiveData.observe(fragment) { result ->
+      processEphemeralStateResult(result)
+    }
+  }
+
+  private fun processEphemeralStateResult(result: AsyncResult<EphemeralState>) {
+    when (result) {
+      is AsyncResult.Failure -> {
+      }
+//        oppiaLogger.e("StateFragment", "Failed to retrieve ephemeral state", result.error)
+      is AsyncResult.Pending -> {
+      } // Display nothing until a valid result is available.
+      is AsyncResult.Success -> processEphemeralState(result.value)
+    }
+  }
+
+  private fun processEphemeralState(ephemeralState: EphemeralState) {
+    if (!ephemeralState.showContinueButtonAnimation) return
+    if (!ephemeralState.hasPreviousState) {
```

Comment:
Why can't this logic be in the ExplorationProgressController? It seems more complicated to have some of the conditions to show the animation there, and some of them here. It should be cleaner just to have the view show the animation when instructed and otherwise require no additional logic.

Reply:
does exploration progress controller have access to the ephemeral state? I suppose state deck has it, but we arent injecting it into the continue button view

Reply:
done -- it was added to the function where ephemeralState is being initialized